src/entities/conditional-logic/model/conditions.ts

```diff
@@ -74,3 +78,226 @@ export const doesNotIncludeValue = <TItem>(
 
   return !input.includes(value);
 };
+
+export const isGreaterThanDate = (input: string, date: string) => {
+  return input ? isAfter(parseISO(input), parseISO(date)) : false;
+};
+
+export const isLessThanDate = (input: string, date: string) => {
+  return input ? isBefore(parseISO(input), parseISO(date)) : false;
+};
+
+export const isEqualToDate = (input: Maybe<string>, date: string) => {
+  if (!input) return false;
+  return input ? isEqual(parseISO(input), parseISO(date)) : false;
+};
+
+export const isBetweenDates = (
+  input: Maybe<string>,
+  minDate: string,
+  maxDate: string,
+) => {
+  return input
+    ? !isBefore(parseISO(input), parseISO(minDate)) &&
+        !isAfter(parseISO(input), parseISO(maxDate))
+    : false;
+};
+
+export const isNotEqualToDate = (input: Maybe<string>, date: string) => {
+  return input ? !isEqual(parseISO(input), parseISO(date)) : false;
+};
+
+export const isOutsideOfDates = (
+  input: Maybe<string>,
+  minDate: string,
+  maxDate: string,
+) => {
+  return input
+    ? isBefore(parseISO(input), parseISO(minDate)) ||
+        isAfter(parseISO(input), parseISO(maxDate))
+    : false;
+};
+
+const getTimeBasedOnFieldName = (
+  fieldName: string,
+  timeRange: { startTime: HourMinute | null; endTime: HourMinute | null },
+): HourMinute | null => {
+  return fieldName === 'from' ? timeRange.startTime : timeRange.endTime;
+};
+
+export const isGreaterThanTimeRange = (
+  timeRange: { startTime: HourMinute | null; endTime: HourMinute | null },
+  { time, fieldName }: { time: HourMinute; fieldName: string },
+): boolean => {
+  const selectedTime = getTimeBasedOnFieldName(fieldName, timeRange);
+  return selectedTime
+    ? timeToMinutes(selectedTime) > timeToMinutes(time)
+    : false;
+};
+
+export const isLessThanTimeRange = (
+  timeRange: { startTime: HourMinute | null; endTime: HourMinute | null },
+  { time, fieldName }: { time: HourMinute; fieldName: string },
+): boolean => {
+  const selectedTime = getTimeBasedOnFieldName(fieldName, timeRange);
+  return selectedTime
+    ? timeToMinutes(selectedTime) < timeToMinutes(time)
+    : false;
+};
+
+export const isEqualToTimeRange = (
+  timeRange: { startTime: HourMinute | null; endTime: HourMinute | null },
+  { time, fieldName }: { time: HourMinute; fieldName: string },
+): boolean => {
+  const selectedTime = getTimeBasedOnFieldName(fieldName, timeRange);
+  return selectedTime
+    ? timeToMinutes(selectedTime) == timeToMinutes(time)
```

Comment:
We could do: 
```suggestion
    ? timeToMinutes(selectedTime) === timeToMinutes(time)
```

To get rid of the ESLint warning